assignment/impl/src/java/org/sakaiproject/assignment/impl/GradeSheetExporter.java

```diff
@@ -432,6 +443,259 @@ public Optional<Workbook> getGradesSpreadsheet(final String query) {
+                                } else if (submission.getSubmitted() && submission.getDateSubmitted() != null) {
+                                    // submitted, but no grade available yet
+                                    objects.set(index, rb.getString("gen.nograd"));
+                                }
+                            } // if
+                        }
+
+                    } else {
+                        AssignmentSubmissionSubmitter[] submissionSubmitters = submission.getSubmitters().toArray(new AssignmentSubmissionSubmitter[]{});
+
+                        if (submissionSubmitters.length == 0) {
+                            continue;
+                        }
+                        Submitter submitter = submitterMap.get(submissionSubmitters[0].getSubmitter());
+                        if (submitter != null) {
+                            // Get the user ID for this result
+                            if (isAnonymousGrading) {
+                                submitter = new Submitter(submission.getId() + " " + rb.getString("grading.anonymous.title"), submitter);
+                            }
+
+                            List<Object> objects = results.computeIfAbsent(submitter, k -> new ArrayList<>(Collections.nCopies(assignmentSize, null)));
+                            // Create item and fill up if doesn't exist.
+                            // find right row
+
+                            if (submission.getGraded() && submission.getGrade() != null) {
+                                // graded and released
+                                String grade = assignmentService.getGradeForSubmitter(submission, submissionSubmitters[0].getSubmitter());
+
+                                if (assignmentTypeOfGrade == Assignment.GradeType.SCORE_GRADE_TYPE) {
+                                    try {
+                                        // numeric cell type?
+                                        int factor = submission.getAssignment().getScaleFactor();
+                                        int dec = (int) Math.log10(factor);
+
+                                        //We get float number no matter the locale it was managed with.
+                                        NumberFormat nbFormat = formattedText.getNumberFormat(dec, dec, null);
+                                        double f = nbFormat.parse(grade).doubleValue();
+
+                                        String format = "#,##0.";
+                                        for (int j = 0; j < dec; j++) {
+                                            format = format.concat("0");
+                                        }
+                                        style.setDataFormat(wb.createDataFormat().getFormat(format));
+                                        cell.setCellStyle(style);
+                                        objects.set(index, new FloatCell(format, f));
+                                    } catch (Exception e) {
```

Comment:
you might be able to use the service method to get a properly displayed score

Reply:
Changes done on this commit:
https://github.com/sakaiproject/sakai/pull/12651/commits/7e35127a9e99d6ba8baf0c434c92527355caf08f